### ISSUE TITLE
Unused arguments in documentation for Mutect2

### ISSUE THREAD

**Issue by meganshand** [broadinstitute/gatk#5352]
### Tool(s) or class(es) involved
Mutect2 and FilterMutectCalls

### Description 
Because both `M2ArgumentCollection` and `M2FiltersArgumentCollection` extend `AssemblyBasedCallerArgumentCollection`, both `Mutect2` and `FilterMutectCalls` display all assembly and caller arguments in the documentation/help even if those arguments don't actually do anything. For example both tools have the argument `--contamination-fraction-to-filter` which has the description:

```
Fraction of contamination in sequencing data (for all samples) to aggressively remove.

If this fraction is greater is than zero, the caller will aggressively attempt to remove contamination 
through biased down-sampling of reads. Basically, it will ignore the contamination fraction of reads for 
each alternate allele. So if the pileup contains N total bases, then we will try to remove 
(N * contamination fraction) bases for each alternate allele.
```

This argument definitely doesn't do anything in `FilteMutectCalls` but I also don't think it's hooked up to do anything in `Mutect2` either (at least when I tried giving it a high value I still got the same calls). This is by design because Mutect has other ways of handling contamination, but the argument is still displayed in both tools' documentation which is confusing. There are other arguments that have the same issue where it's unclear if they do anything in Mutect or not.

**Comment by droazen** [comment]
If only `HaplotypeCaller` uses `--contamination-fraction-to-filter`/`--contamination-fraction-per-sample-file`, then perhaps those arguments should be moved down into `HaplotypeCallerArgumentCollection`?

**Comment by davidbenjamin** [comment]
Fixed for filtering in #5498.  `Mutect2` itself will be trickier because a lot of unused arguments are a result of `SomaticGenotypingEngine`'s inheritance.  This will involve significant refactoring, but the genotyping classes will make more sense.
